Vizzuality/half-earth-v3
# Fix/edit aoi name functionality

## Add edit name functionality to AOI view
### Description

- The edit name of the area was not implemented on the AOI view. Now is similar to the one on the AOI list.
- This button only appears now on the custom areas, is hidden on the rest.

Also, a bug where AOI was not correctly stored if they didn't have any species is fixed. This was causing the reload of the AOI page and the loading of some areas to fail.

### Designs
[_Link to the related design prototypes (if relevant)_
](https://www.figma.com/file/8lZsKUaGM6K9JOIG0orY76/HE---Design-System?node-id=1824%3A47457)
### Testing instructions
Go to one custom AOI. You should be able to change the name. It should change immediately and be stored.
If you go to a preloaded AOI you shouldn't see this button
If you reload the page it should be loaded again
### Feature relevant tickets
[_Link to the related task manager tickets_](https://vizzuality.atlassian.net/browse/HE-250?atlOrigin=eyJpIjoiNmExYjk1YTdiOTE3NDdiNTliMGUyZjE0ZmU0N2RiOTgiLCJwIjoiaiJ9)
